dali/test/python/test_operator_one_hot.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2017-2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nvidia.dali.pipeline import Pipeline
+import nvidia.dali.ops as ops
+import nvidia.dali.types as types
+
+import numpy as np
+
+sample_size = 10
+premade_batch = [np.array(x, dtype=np.int32) for x in range(sample_size)]
+
+
+class OneHotPipeline(Pipeline):
+    def __init__(self, nclasses, num_threads=1):
+        super(OneHotPipeline, self).__init__(sample_size,
+                                             num_threads,
+                                             0)
+        self.ext_src = ops.ExternalSource()
+        self.one_hot = ops.OneHot(depth=nclasses, dtype=types.INT32, device="cpu")
```

Comment:
You can test more types.

Reply:
I was not able to add such python test because it would require a np.type -> python.type mapping (operator uses python.types lib). But I added a unit typed_test that runs different type scenarios in cpp

Reply:
You can use this types to create a numpy array you seed to the ExternalSource - so it defines your input type.
To go through output types you can use `nvidia.dali.types.DALIDataType.__members__.values() `

Reply:
Also in test you are testing kernel, nad for sanity it would be good to test operator as well.